### DISCUSSION TITLE
Yank relative/absolute path to a document

### DISCUSSION THREAD

**Discussion by pawelgrzybek** [helix-editor/helix#10679] [General]
Hi.

Currently there is no easy way to copy a path to a current document. The way how I do it now is `:sh echo <ctrl-r %> | pbcopy` which gives me an absolute path to a current document in my clipboard.  I think it is a useful to have it as an individual command. Use case ideas:

1. Copy a path, open a new terminal window, `touch` a new file at the same location.
2. Copy a path, open a new terminal window, `git blame` a file

Implementation ideas:

- New command `:yank-path-absolute` / `:ypa`
- New command `:yank-path-relative` / `:ypr`

I am not a very proficient Rust dev, but if you like the idea I am more than happy to give it a go and implement it. Let me know your thoughts.

**Comment by kirawi** [comment]
I think the plan is to have https://github.com/helix-editor/helix/issues/9552 address this use case

**Comment by softkot** [comment]
Since DAP support is still deeply experimental and in its initial state, debugging is still convenient to carry out in external tools such as lldb or dlv, and for convenience, of course, there is a strong lack of the ability to configure hotkeys to copy the current path and line number to quickly set up break points in debug tools.

Actually i thouhgt this will works, but no luck 
```
[keys.normal]
s = ':sh echo b #{filename}:#{linenumber} | xclip -i'

```
